dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.Indices#flushIndex(org.springframework.http.HttpHeaders, org.elasticsearch.action.admin.indices.flush.FlushRequest)
+     */
+    @Override
+    public Mono<Void> flushIndex(HttpHeaders headers, FlushRequest flushRequest) {
+
+        return sendRequest(flushRequest, DotReactiveElasticsearchClient.RequestCreator.flushIndex(), FlushResponse.class, headers) //
+                .then();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient#ping(org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient.ReactiveElasticsearchClientCallback)
+     */
+    @Override
+    public Mono<ClientResponse> execute(ReactiveElasticsearchClientCallback callback) {
+
+        return this.hostProvider.getActive(Verification.LAZY) //
+                .flatMap(callback::doWithClient) //
+                .onErrorResume(throwable -> {
+
+                    if (throwable instanceof ConnectException) {
+
+                        return hostProvider.getActive(Verification.ACTIVE) //
+                                .flatMap(callback::doWithClient);
+                    }
+
+                    return Mono.error(throwable);
+                });
+    }
+
+    @Override
+    public Mono<Status> status() {
+
+        return hostProvider.clusterInfo() //
+                .map(it -> new DotReactiveElasticsearchClient.ClientStatus(it.getNodes()));
+    }
+
+    // --> Private Response helpers
+
+    private static GetResult getResponseToGetResult(GetResponse response) {
+
+        return new GetResult(response.getIndex(), response.getType(), response.getId(), response.getSeqNo(),
+                response.getPrimaryTerm(), response.getVersion(), response.isExists(), response.getSourceAsBytesRef(),
+                response.getFields(), null);
+    }
+
+    // -->
+
+    private <Req extends ActionRequest, Resp> Flux<Resp> sendRequest(Req request, Function<Req, Request> converter,
+            Class<Resp> responseType, HttpHeaders headers) {
+        return sendRequest(converter.apply(request), responseType, headers);
+    }
+
+    private <Resp> Flux<Resp> sendRequest(Request request, Class<Resp> responseType, HttpHeaders headers) {
+
+        String logId = ClientLogger.newLogId();
+
+        return execute(webClient -> sendRequest(webClient, logId, request, headers))
+                .flatMapMany(response -> readResponseBody(logId, request, response, responseType));
+    }
+
+    private Mono<ClientResponse> sendRequest(WebClient webClient, String logId, Request request, HttpHeaders headers) {
+
+        RequestBodySpec requestBodySpec = webClient.method(HttpMethod.valueOf(request.getMethod().toUpperCase())) //
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'requestBodySpec' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)